src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -226,6 +229,13 @@ private async Task ProcessIncomingFramesAsync()
                     frameHeader = await ReadFrameAsync().ConfigureAwait(false);
                     if (NetEventSource.IsEnabled) Trace($"Frame {frameNum}: {frameHeader}.");
 
+                    if (_lastStreamId != -1 && frameHeader.StreamId > _lastStreamId)
```

Comment:
I'm not 100% sure if there are any frames which can be handled better than ignore. I.e. not sure if new stream request (FrametType.Headers) should be ignored or handled and explicitly shutdown (so basically go into the same code path as for _dispose)

Reply:
Per spec:

```
If the receiver of the GOAWAY has sent data on streams with a higher stream identifier than what is indicated in the GOAWAY frame, those streams are not or will not be processed.
```

but I'm not actually sure this is always ok (i.e. new stream request might be better if we responded as client could have not received GOAWAY yet when they've sent a frame)

Reply:
Do we even need this logic at all? Once we receive GOAWAY, all streams above _lastStreamId should be removed. And we already ignore frames on streams that are closed.

Reply:
Removed